src/messaging/ExchangeMgr.h

```diff
@@ -0,0 +1,255 @@
+     *                                                             is full and a new one cannot be allocated.
+     *  @retval #CHIP_NO_ERROR On success.
+     */
+    CHIP_ERROR RegisterUnsolicitedMessageHandler(uint32_t protocolId, uint8_t msgType, ExchangeContext::MessageReceiveFunct handler,
+                                                 void * appState);
+
+    /**
+     *  Register an unsolicited message handler for a given profile identifier and message type.
+     *
+     *  @param[in]    protocolId     The profile identifier of the received message.
+     *
+     *  @param[in]    msgType       The message type of the corresponding profile.
+     *
+     *  @param[in]    handler       The unsolicited message handler.
+     *
+     *  @param[in]    allowDups     Boolean indicator of whether duplicate messages are allowed for a given
+     *                              profile identifier and message type.
+     *
+     *  @param[in]    appState      A pointer to a higher layer object that holds context state.
+     *
+     *  @retval #CHIP_ERROR_TOO_MANY_UNSOLICITED_MESSAGE_HANDLERS If the unsolicited message handler pool
+     *                                                             is full and a new one cannot be allocated.
+     *  @retval #CHIP_NO_ERROR On success.
+     */
+    CHIP_ERROR RegisterUnsolicitedMessageHandler(uint32_t protocolId, uint8_t msgType, ExchangeContext::MessageReceiveFunct handler,
+                                                 bool allowDups, void * appState);
+
+    /**
+     *  Unregister an unsolicited message handler for a given profile identifier.
+     *
+     *  @param[in]    protocolId     The profile identifier of the received message.
+     *
+     *  @retval #CHIP_ERROR_NO_UNSOLICITED_MESSAGE_HANDLER  If the matching unsolicited message handler
+     *                                                       is not found.
+     *  @retval #CHIP_NO_ERROR On success.
+     */
+    CHIP_ERROR UnregisterUnsolicitedMessageHandler(uint32_t protocolId);
+
+    /**
+     *  Unregister an unsolicited message handler for a given profile identifier and message type.
+     *
+     *  @param[in]    protocolId     The profile identifier of the received message.
+     *
+     *  @param[in]    msgType       The message type of the corresponding profile.
+     *
+     *  @retval #CHIP_ERROR_NO_UNSOLICITED_MESSAGE_HANDLER  If the matching unsolicited message handler
+     *                                                       is not found.
+     *  @retval #CHIP_NO_ERROR On success.
+     */
+    CHIP_ERROR UnregisterUnsolicitedMessageHandler(uint32_t protocolId, uint8_t msgType);
+
+private:
+    uint16_t NextExchangeId;
+    uint8_t mState;                     /**< [READ ONLY] The state of the ExchangeManager object. */
+    SecureSessionMgrBase * mSessionMgr; /**< [READ ONLY] The associated SecureSessionMgrBase object. */
+
+    class UnsolicitedMessageHandler
```

Comment:
a class  with only public data members may be better represented as a struct.

Reply:
Move all members to private

Reply:
Does this need an update? the new code still shows a class with all members public.